include/souper/Infer/Pruning.h

```diff
@@ -60,6 +60,7 @@ class PruningManager {
   HoleAnalysis HA;
   llvm::KnownBits LHSKnownBitsNoSpec;
   InputVarInfo LHSMustDemandedBits;
+  bool EnableDemandedBitsPruning = false;
```

Comment:
does this really want to be a state variable, as opposed to a parameter that gets passed around?

Reply:
I think state variable is appropriate, because it is not intended for an user.
The pruning initialization function sets it to true if the 'must demanded' analysis of the LHS returns an useful result.